src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Windows.cs

```diff
@@ -80,6 +80,7 @@ private unsafe bool InitCultureDataCore()
             // Start by assuming the windows name will be the same as the specific name since windows knows
             // about specifics on all versions. Only for downlevel Neutral locales does this have to change.
             _sWindowsName = realNameBuffer;
+            _isDefaultUserLocale = _sWindowsName == CultureInfo.GetUserDefaultLocaleName();
```

Comment:
>CultureInfo.GetUserDefaultLocaleName(); [](start = 52, length = 39)

should we use the cached value s_userDefaultLocaleName instead? I think we shouldn't call the OS every time we create a culture.

Reply:
Will it call the OS everytime? `CultureInfo.GetUserDefaultLocaleName` is using `Interlocked.CompareExchange` and only calling the OS when `s_userDefaultLocaleName == null`.

Reply:
also we need to do that only if useroverride is true. that is why I was preferring to have just one variable encapsulate the logic.

---
In reply to: [445776569](https://github.com/dotnet/runtime/pull/38372#discussion_r445776569) [](ancestors = 445776569)

Reply:
> also we need to do that only if useroverride is true

You mean, only do the caching if `userOverride` is true, or doing the comparison at all?

Reply:
I mean encapsulate the userOverride and _isDefaultUserLocale to only one variable. and this check will be like 

```C#
userOverride = userOverride || _sWindowsName == CultureInfo.GetUserDefaultLocaleName();
```

or something like that.

Reply:
> Will it call the OS everytime? CultureInfo.GetUserDefaultLocaleName is using Interlocked.CompareExchange and only calling the OS when s_userDefaultLocaleName == null.

That will be too much logic calling compare exchange every time you need to access the variable. I don't think we need to do that. caching the value in the initialized static variable should be enough and simpler.

Reply:
Makes sense... working on that change now